Quadriga/src/main/java/edu/asu/spring/quadriga/web/transformation/TransformationController.java

```diff
@@ -0,0 +1,88 @@
+package edu.asu.spring.quadriga.web.transformation;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.spring.quadriga.domain.impl.networks.Transformation;
+import edu.asu.spring.quadriga.domain.network.tranform.ITransformation;
+import edu.asu.spring.quadriga.exceptions.FileStorageException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaGeneratorException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.network.transform.impl.Generator;
+import edu.asu.spring.quadriga.service.network.transform.impl.MatchGraphs;
+import edu.asu.spring.quadriga.service.network.transform.impl.TransformNode;
+import edu.asu.spring.quadriga.service.transformation.ITransformationManager;
+
+@Controller
+public class TransformationController {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransformationController.class);
+
+    @Autowired
+    private MatchGraphs matchGraphs;
+
+    @Autowired
+    private ITransformationManager transformationManager;
+
+    @Autowired
+    private Generator generator;
+
+    /**
+     * Gets transformation-request,networks from the User and returns
+     * transformed graph Extensions: Obtain Mapping and Pattern files from the
+     * user.
+     * 
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws FileStorageException
+     */
+
+    @RequestMapping(value = "checks/transformation", method = RequestMethod.POST)
+    public String transformTest(@RequestParam("project") String project,
+            @RequestParam("transformation") String transformationNameAndID,
```

Comment:
do you need the transformation name for anything? if not, you don't need to submit it (especially not as combined string with the id).